### PR TITLE
Select menu for mime type, driven by authority config files

### LINKED ISSUE #87: File format dropdown
Add dropdown menu on the fileset screen to select a file format. Pass this list via a hidden element or a data attribute.

### PR DESCRIPTION
**Pull request by escowles**
* Removing `conforms_to` in favor of `mime_type` values with extra attributes to differentiate different XML-based metadata formats

Fixes #87

### CODE REVIEW

File: config/authorities/raster_formats.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+terms:
+  - id: image/tiff; gdal-format=GTiff
+    term: GeoTIFF
+  - id: text/plain; gdal-format=AIGrid
+    term: AIGrid
```
Comment (eliotjordan) [inline config/authorities/raster_formats.yml]: Should be: 
  - id: text/plain; gdal-format=AAIGrid
  - term: Arc/Info ASCII Grid

We should also add: 
- id: application/octet-stream; gdal-format=AIG
- term: Arc/Info Binary Grid

File: config/authorities/vector_formats.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+terms:
+  - id: application/zip; ogr-format="ESRI Shapefile"
+    term: ESRI Shapefile
+  - id: application/zip; ogr-format=OpenFileGDB
+    term: OpenFileGDB
+  - id: application/octet-stream; ogr-format=MDB
+    term: MDB
```
Comment (eliotjordan) [inline config/authorities/vector_formats.yml]: Let's remove MDB, it's not a default format in GDAL.

File: config/authorities/image_formats.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+terms:
+  - id: image/jpeg
+    term: JPEG
+  - id: image/tiff
+    term: TIFF
+  - id: application/octet-stream; ogr-format=MDB
```
Comment (eliotjordan) [inline config/authorities/image_formats.yml]: This is in the wrong section. Should be deleted.

**Comment by eliotjordan** [comment]
@drh @johnhuck @jrgriffiniii @yulgit1 what do you think of this [approach](https://github.com/projecthydra-labs/geo_concerns/blob/mime-select/config/authorities/metadata_formats.yml)? It seems reasonable to indicate FileSet type for external metadata files in the same way as vector, raster, and image file sets - using mime type with a parameter instead of conformsTo. I think having two separate mechanisms for this is overly complex.

**Comment by drh-stanford** [comment]
That's fine with me. MODS already has the MIME type `application/mods+xml`

File: config/authorities/metadata_formats.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+terms:
+  - id: application/xml; schema=fgdc
+    term: FGDC
+  - id: application/xml; schema=iso19139
+    term: ISO19139
+  - id: application/xml; schema=mods
```
Comment (eliotjordan) [inline config/authorities/metadata_formats.yml]: MODS has an existing MIME type: application/mods+xml

Comment (escowles) [inline config/authorities/metadata_formats.yml]: I found a couple instances of `application/fgdc+xml`, should we go with that?  Maybe `application/iso19139+xml` too?

**Comment by eliotjordan** [comment]
@escowles In regards to metadata MIME types, MODS is the only [officially registered type](https://tools.ietf.org/html/rfc6207) in the group. I don't think we should be using `application/*+xml` for all of them.

**Comment by jpstroop** [comment]
What would you do instead? I don't think I'd recommend the `x-` syntax:

https://tools.ietf.org/html/rfc6838#section-3.4

Maybe better to go with what the assumed media type would be, and then start the registration process? I don't think it's too difficult.

**Comment by escowles** [comment]
The previous versions were `application/xml; schema=iso19139` and `application/xml; schema=fgdc`.  The `schema=` part is completely made up, but follows the `ogr-format` and `gdal-format` pattern used for the other formats.

If you google "application/fgdc+xml", you'll see a few hits, though it isn't registered at https://www.iana.org/assignments/media-types .

**Comment by eliotjordan** [comment]
@jpstroop We thought about that, but the issue is the same with geospatial data formats. That's why the data modeling group settled on noting the file format with a parameter. Except in cases where there is an existing registered type.

**Comment by eliotjordan** [comment]
I'm not sure of the rules, but it seems reasonable to me that the group responsible for creating and maintaining standards should register the mime type.

**Comment by jpstroop** [comment]
@eliotjordan yes. We did this for MODS and MARC, etc. when I was on the MODS editorial committee:

https://tools.ietf.org/html/rfc6207

I'd definitely recommend pressuring whoever is in charge of these GIS specs to get them registered. I don't recall it being difficult.

**Pull request merged**